client-react/src/ApiHelpers/HttpClient.ts

```
@@ -37,3 +38,30 @@ export const getTextHeaders = (): KeyValue<string> => {
     Accept: 'text/plain,*/*',
   };
 };
+
+export const getLinksFromLinkHeader = (linksHeader: string): { [key: string]: string } => {
+  const links: { [key: string]: string } = {};
+
+  if (linksHeader) {
+    // Parse each part into a named link
+    linksHeader.split(',').forEach(part => {
+      const section = part.split(';');
+      const url = section[0].replace(/<(.*)>/, '$1').trim();
+      const name = section[1].replace(/rel="(.*)"/, '$1').trim();
+      links[name] = url;
+    });
+  }
+
+  return links;
+};
+
+export const getLastPageNumberFromLinks = (links: { [key: string]: string }) => {
+  const lastPageLink = links && links.last;
+  if (lastPageLink) {
+    const lastPageNumberString = Url.getParameterByName(lastPageLink, 'page');
+    if (lastPageNumberString) {
+      return +lastPageNumberString;
```

Umm I don't think I have seen this before, but what is this `+` used for?

---

This converts the value into a number

---

we use it some places... basically converts a string to number. In this case worst case the page number returned by GitHub is a messed up string rather than a number, in which case, this will return a NaN and the following call will fail. Which will be logged. We dont want to throw any errors here.